demo/runtime/src/lib.rs

```diff
@@ -176,6 +176,12 @@ impl_outer_event! {
 	}
 }
 
+impl_outer_log! {
+	pub enum Log for Concrete {
+		consensus(AuthoritiesChange => consensus::AuthoritiesChange<SessionKey> => as_authorities_change)
```

Comment:
consider just introducing `trait AsAuthoritiesChange { fn as_authorities_change(self) -> Result<Vec<SessionKey>, Self> }` and then manually making an `impl` of it here. might save quite a lot of somewhat opaque syntax.

Reply:
Substrate has an access to digest items only through `DigestItem` trait, which is shared among all possible runtimes (declared in primitives).  Extracting `as_authorities_change` (and other similar conversions later) into separate conversion trait would require `DiggestItem` to inherit from all those traits + will make it impossible to use default implementation (which returns `None` right now) => all runtimes have to explicitly declare support even for those 'system' digest items, which they are not willing to support (by simply using `StubDigestItem` and returning `None`).

I'm not sure it is better - this is an additional overhead for runtimes development. From the other side, it will force runtime developers to explicitly throw away items the do not want to support. Left as is for now, but could extract if you insist.

Reply:
Probably also could remove `AuthoritiesChangeDigest` trait in favor of `Vec<AuthorityId>`.